src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,794 @@
+
+        # Is the input datatype signed or unsigned?
+        # The thresholding core needs to know this when comparing weights to inputs
+        if self.get_input_datatype().signed():
+            code_gen_dict["$SIGNED$"] = [str(1)]
+        else:
+            code_gen_dict["$SIGNED$"] = [str(0)]
+
+        if bias >= 0:
+            o_bits = math.ceil(math.log2(2**o_bitwidth + bias))
+        else:
+            o_bits = 1 + math.ceil(
+                math.log2(-bias if -bias >= 2 ** (o_bitwidth - 1) else 2**o_bitwidth + bias)
+            )
+
+        code_gen_dict["$O_BITS$"] = [str(int(o_bits))]
+
+        rt_weights = self.get_nodeattr("runtime_writeable_weights")
+        code_gen_dict["$USE_AXILITE$"] = [str(rt_weights)]
+
+        depth_trigger_uram = self.get_nodeattr("depth_trigger_uram")
+        depth_trigger_bram = self.get_nodeattr("depth_trigger_bram")
+        deep_pipeline = self.get_nodeattr("deep_pipeline")
+        code_gen_dict["$DEPTH_TRIGGER_URAM$"] = [str(depth_trigger_uram)]
+        code_gen_dict["$DEPTH_TRIGGER_BRAM$"] = [str(depth_trigger_bram)]
+        code_gen_dict["$DEEP_PIPELINE$"] = [str(deep_pipeline)]
+        return code_gen_dict
+
+    def get_rtl_file_list(self):
+        """Thresholding binary search RTL file list"""
+        return [
+            "axilite_if.v",
+            "thresholding.sv",
+            "thresholding_axi.sv",
+            "thresholding_template_wrapper.v",
+        ]
+
+    def get_rtl_file_paths(self):
+        """Get full path of all RTL files"""
+        rtl_root_dir = os.environ["FINN_ROOT"] + "/finn-rtllib/thresholding/hdl/"
+        rtl_file_list = self.get_rtl_file_list()
+        rtl_file_paths = [rtl_root_dir + file for file in rtl_file_list]
+        return rtl_file_paths
+
+    def get_rtl_template_data(self, path):
+        """Return RTL file contents as a template"""
+        with open(path, "r") as f:
+            template = f.read()
+        return template
+
+    def fill_in_rtl_template_data(self, replace_dict, template_data):
+        """Use attribute values to finn in RTL template placeholders"""
+        template_data_cp = template_data
+        for key in replace_dict:
+            replacement_line = "\n".join(replace_dict[key])
+            template_data_cp = template_data_cp.replace(key, replacement_line)
+        return template_data_cp
+
+    def dump_rtl_data(self, dest_dir, filename, data):
+        """Dump filled-in-template RTL files for future synthesis step"""
+        # when generating template files, handle a special case:
+        # if the filename contains the word "template", replace that
+        # with the node name to distinguish between instances
+        filename = filename.replace("template", self.onnx_node.name)
+        with open(os.path.join(dest_dir, filename), "w") as f:
+            f.write(data)
+        return
+
+    def code_generation_ipgen(self, model, fpgapart, clk):
```

Comment:
This is already defined in `RTLBackend` and doesn't need to be redefined: https://github.com/Xilinx/finn/blob/c4e57da733bff97d63daa8692fc08a05f2d1ba59/src/finn/custom_op/fpgadataflow/rtlbackend.py#L56
So this function can be removed.